### ISSUE TITLE
separate out "cached file/map" generation from staleness checks

### ISSUE THREAD

**Issue by sandfox** [trello/node-coffee-cache#10]
Would be cool if the logic/code for creating the compiled JS+map was in a seperate method, as opposed to the single method `cacheFile` now which both checks the staleness of the cached files and then optionally recompiles them.

I'll try and do a PR for this, but just wanted to check it's not a daft idea. My use case is "force" updating the cache for a single files without resorting to removing the entire cache dir.

**Comment by dpatti** [comment]
Yeah, I think it's a great idea. I started on a 2.x branch that does more breaking apart like you want, but I also haven't pushed to it in a bit. I would certainly accept a pull request off master for what you're trying to do.

**Comment by sandfox** [comment]
Ok, will see what I can for the current 1.x branch.

On a tangent (and probably the wrong place for it), I noticed you've introduced peerDeps on 2.x, are you aware of the current desire to deprecate that feature -> https://github.com/npm/npm/issues/5080#issuecomment-40545599
(yeah it's a long thread but I think I've linked to a good point in it)

**Comment by dpatti** [comment]
I did _not_ know about that, and I'll probably remove it. Thanks for the link!

**Comment by alexlawrence** [comment]
+1
